lib/output-files.js

```diff
@@ -1,19 +1,60 @@
 // output JavaScript bundled in puppeteer output to format
 // that can be eaten by Istanbul.
+
+// TODO: Put function interfaces on this file
+
+const fs = require('fs')
+const mkdirp = require('mkdirp')
+const clone = require('clone')
+
 class OutputFiles {
   constructor (coverageInfo) {
-    this.coverageInfo = coverageInfo
+    // Clone coverageInfo to prevent mutating the passed in data
+    this.coverageInfo = clone(coverageInfo)
+    this.iterator = 0
+    this.parseAndIsolate()
   }
+
   rewritePath (path) {
     // generate a new path relative to ./coverage/js.
     // this would be around where you'd use mkdirp.
-    return `${path}-hey!`
+
+    var str = ``
+
+    // Get the last element in the path name
+    var truncatedPath = path.split('/')[path.split('/').length - 1]
```

Comment:
you should be able to use [path.basename()](https://nodejs.org/api/path.html#path_path_basename_path_ext) for this logic -- it does pretty much the same thing I believe.